File: solutions/partition-array-according-to-given-pivot.py
from typing import List


class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        left = []
        middle = []
        right = []

        for num in nums:
            if num > pivot:
                right.append(num)
            elif num < pivot:
                left.append(num)
            else:
                middle.append(num)

        return left + middle + right

File: solutions/partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int[] result = new int[nums.length];
        int lessIndex = 0;
        int greaterIndex = nums.length - 1;

        for (int i = 0, j = nums.length - 1; i < nums.length; i++, j--) {
            if (nums[i] < pivot) {
                result[lessIndex] = nums[i];
                lessIndex++;
            }

            if (nums[j] > pivot) {
                result[greaterIndex] = nums[j];
                greaterIndex--;
            }
         }

        while (lessIndex <= greaterIndex) {
            result[lessIndex] = pivot;
            lessIndex++;
        }

        return result;
    }
}
